src/plugins/intel_cpu/src/nodes/executors/shl/shl.hpp

```diff
@@ -0,0 +1,199 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+
+#include "shl_utils.hpp"
+#include "csinn/csinn_data_structure.h"
+#include "csinn/csinn_runtime.h"
+
+#include "memory_desc/cpu_memory_desc.h"
+
+#include <memory>
+
+
+namespace ov {
+namespace intel_cpu {
+
+
+template <typename T>
+struct ShlStructureTraits {};
+
+template <typename T, typename traits = ShlStructureTraits<T>>
+struct ShlStructure {
+public:
+    ShlStructure() = default;
+    ShlStructure(const ShlStructure<T, traits>&) = default;
+    ShlStructure(ShlStructure<T, traits>&&) = default;
+    explicit ShlStructure(T t) { reset(t); }
+
+    ShlStructure<T, traits> &operator=(const ShlStructure<T, traits>&) = default;
+    ShlStructure<T, traits> &operator=(ShlStructure<T, traits>&&) = default;
+
+    void reset(T t) {
+        m_ptr.reset(t, traits::destructor);
+    }
+
+    T get(bool allow_empty = false) const {
+        T result = m_ptr.get();
+        OPENVINO_ASSERT(allow_empty || result != nullptr, "ShlStructure is not initialized");
+        return result;
+    }
+
+    explicit operator T() const {
+        return get(true);
+    }
+
+    explicit operator bool() const {
+        return get(true) != nullptr;
+    }
+
+    bool operator==(const ShlStructure<T, traits> &other) const {
+        return other.m_ptr.get() == m_ptr.get();
+    }
+    bool operator!=(const ShlStructure &other) const {
+        return !(*this == other);
+    }
+
+private:
+    std::shared_ptr<typename std::remove_pointer<T>::type> m_ptr = nullptr;
+
+protected:
+    bool operator==(const T other) const { return other == m_ptr.get(); }
+    bool operator!=(const T other) const { return !(*this == other); }
+};
+
+template <>
+struct ShlStructureTraits<csinn_session*> {
+    static void destructor(csinn_session* p) {
+        return csinn_free_session(p);
+    }
+};
```

Comment:
Is it possible to run valgrind or sanitizer on RISCV?
Just to double check that we don't leak